app/controllers/addresses_controller.rb

```diff
@@ -9,9 +9,15 @@ def new
   end
   
   def create
-    @restaurant = Restaurant.find(params[:restaurant_id])
-    @address = Address.create(address_params)
-    redirect_to restaurants_path(@restaurant)
+    restaurant = Restaurant.find(params[:restaurant_id])
+    address = restaurant.addresses.build(address_params)
+    if address.valid?
```

Comment:
you could also do `if address.save` here, the `save` method will return false if the address is not saved to the database. Then on line `18` you can display the errors to the user

```
redirect_to root_path, , alert: address.errors.full_messages.join(", ")
```